### PR TITLE
Added optional support for static libvpx and the yasm compiler.

### PR DESCRIPTION
**Pull request by ggarra13**
* Added optional FFmpeg flags.
* Fixed non-official repository of zlib.

### CODE REVIEW

**Comment by ggarra13** [comment]
A word of cauition.  The VPX configure script looks for yasm in the user's PATH as it does not use absolute paths.

File: etc/SuperBuild/CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ set(TLRENDER_TIFF TRUE CACHE BOOL "Enable support for TIFF")
 set(TLRENDER_PNG TRUE CACHE BOOL "Enable support for PNG")
 set(TLRENDER_EXR TRUE CACHE BOOL "Enable support for OpenEXR")
 set(TLRENDER_FFMPEG TRUE CACHE BOOL "Enable support for FFmpeg (see the README for how to build FFmpeg on Windows)")
+set(TLRENDER_VPX TRUE CACHE BOOL "Enable support for VPX codec thru FFmpeg")
```
Comment (darbyjohnston) [inline etc/SuperBuild/CMakeLists.txt:19]: Typo: thru -> through

File: etc/SuperBuild/CMakeLists.txt [hunk 2]
```diff
@@ -93,6 +94,10 @@ endif()
 if(TLRENDER_EXR)
     list(APPEND TLRENDER_EXTERNAL_DEPS OpenEXR)
 endif()
+if(TLRENDER_VPX AND TLRENDER_FFMPEG AND NOT WIN32)
+    list(APPEND TLRENDER_EXTERNAL_DEPS YASM)
```
Comment (darbyjohnston) [inline etc/SuperBuild/CMakeLists.txt:98]: NASM is already a dependency, can it be used instead of YASM?

File: etc/SuperBuild/cmake/Modules/BuildFFmpeg.cmake [hunk 3]
```diff
@@ -12,12 +12,23 @@ else()
...
         list(APPEND FFmpeg_CFLAGS "--extra-cflags=-mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
         list(APPEND FFmpeg_CXXFLAGS "--extra-cxxflags=-mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
         list(APPEND FFmpeg_OBJCFLAGS "--extra-objcflags=-mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
         list(APPEND FFmpeg_LDFLAGS "--extra-ldflags=-mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
     endif()
+    if(TLRENDER_VPX)
+        list(APPEND FFmpeg_CONFIGURE_ARGS
+            --enable-libvpx)
+        #
+        # Makre sure we pick the static libvpx we compiled, not the system one
```
Comment (darbyjohnston) [inline etc/SuperBuild/cmake/Modules/BuildFFmpeg.cmake:26]: Typo: Makre -> Make

Using CMAKE_PREFIX_PATH won't work here, it can contain multiple paths.

File: etc/SuperBuild/cmake/Modules/BuildVPX.cmake [hunk 4]
```diff
@@ -0,0 +1,37 @@
+include(ExternalProject)
+
+
+# set(LIBVPX_TAG main) # live on the cutting-edge!
+
+set(VPX_TAG v1.12.0) # ptoven to work
```
Comment (darbyjohnston) [inline etc/SuperBuild/cmake/Modules/BuildVPX.cmake:6]: Typo: ptoven -> proven

File: etc/SuperBuild/cmake/Modules/BuildYASM.cmake [hunk 5]
```diff
@@ -0,0 +1,19 @@
+include( ExternalProject )
+
+# Use this for cuting edge yasm
+# GIT_REPOSITORY "https://github.com/yasm/yasm.git"
+
+ExternalProject_Add(
+  YASM
```
Comment (darbyjohnston) [inline etc/SuperBuild/cmake/Modules/BuildYASM.cmake:7]: Indentation is four spaces.

File: etc/SuperBuild/cmake/Modules/BuildZLIB.cmake [hunk 6]
```diff
@@ -10,8 +8,7 @@ set(ZLIB_ARGS
 ExternalProject_Add(
     ZLIB
     PREFIX ${CMAKE_CURRENT_BINARY_DIR}/ZLIB
-    GIT_REPOSITORY ${ZLIB_GIT_REPOSITORY}
-    GIT_TAG ${ZLIB_GIT_TAG}
+    URL "https://zlib.net/fossils/zlib-1.2.11.tar.gz"  # Original repository
```
Comment (darbyjohnston) [inline etc/SuperBuild/cmake/Modules/BuildZLIB.cmake:11]: Why is this change included?

**Review by darbyjohnston — CHANGES_REQUESTED** [review]

File: etc/SuperBuild/CMakeLists.txt (hunk 2 above)
Comment (ggarra13) [reply etc/SuperBuild/CMakeLists.txt:98]: Yes, NASM can be used, but LibVPX recommends YASM for performance.

File: etc/SuperBuild/cmake/Modules/BuildZLIB.cmake (hunk 6 above)
Comment (ggarra13) [reply etc/SuperBuild/cmake/Modules/BuildZLIB.cmake:11]: It is unrelated, but I noticed you were picking ZLIB from a, afaik, non-official distribution.  Also, with that you need to update the tag on every release.  By pointing to the official distro of directory fossils, you can pick any version and update it whenever you see fit.

File: etc/SuperBuild/cmake/Modules/BuildFFmpeg.cmake (hunk 3 above)
Comment (ggarra13) [reply etc/SuperBuild/cmake/Modules/BuildFFmpeg.cmake:26]: Should I rely on CMAKE_INSTALL_PREFIX instead?

File: etc/SuperBuild/CMakeLists.txt (hunk 1 above)
Comment (ggarra13) [reply etc/SuperBuild/CMakeLists.txt:19]: No slang, eh?

File: etc/SuperBuild/CMakeLists.txt (hunk 2 above)
Comment (darbyjohnston) [reply etc/SuperBuild/CMakeLists.txt:98]: I don't really want to have multiple assemblers as dependencies, can the others also use YASM (jpegturbo, ffmpeg)?

File: etc/SuperBuild/cmake/Modules/BuildFFmpeg.cmake (hunk 3 above)
Comment (darbyjohnston) [reply etc/SuperBuild/cmake/Modules/BuildFFmpeg.cmake]: I think that would work but try it out to make sure.

File: etc/SuperBuild/cmake/Modules/BuildZLIB.cmake (hunk 6 above)
Comment (darbyjohnston) [reply etc/SuperBuild/cmake/Modules/BuildZLIB.cmake:11]: The github repository is official, from the zlib home page:
"zlib development contributions and issues are best addressed on [zlib's GitHub repository](https://github.com/madler/zlib)"

File: etc/SuperBuild/CMakeLists.txt (hunk 2 above)
Comment (ggarra13) [reply etc/SuperBuild/CMakeLists.txt]: Not sure.  I left it with NASM as default and YASM as an optional dependency you can turn on.

**Comment by darbyjohnston** [comment]
Hi and sorry for the delay! Hopefully I will have some time next week to test these changes and review the PR.

**Comment by darbyjohnston** [comment]
As I mentioned before, I would really like to keep the dependencies to a minimum and don't like having multiple assemblers. If YASM is a better choice than I think it would be worthwhile to try and replace NASM completely. Looking at the GitHub repos most of the commits to YASM are quite old, while it looks like NASM is under more active development?
